### ISSUE TITLE
Start using mypy (type annotations in python2)

### ISSUE THREAD

**Issue by khenderick** [openvstorage/framework#1037]
http://blog.zulip.org/2016/10/13/static-types-in-python-oh-mypy/

**Comment by wimpers** [comment]
@khenderick Do you want to tackle this while tackling the Swagger API stuff?

**Comment by khenderick** [comment]
No, I want to tackle this when we have some spare time left, it might (or might not) take lots of time to get this right. It's an unknown beast.

**Comment by wimpers** [comment]
in that case set to roadmap

**Comment by JeffreyDevloo** [comment]
Currently implementing this along the way. It provides a way for some static type checking.
Requirement: python-typing package which needs to be converted from pypi.
